descartes_light/include/descartes_light/interface/edge_evaluator.h

```diff
@@ -32,13 +32,59 @@ template <typename FloatType>
 class EdgeEvaluator
 {
 public:
+  using Ptr = typename std::shared_ptr<EdgeEvaluator<FloatType>>;
+  EdgeEvaluator(std::size_t dof) : dof_(dof) {}
   virtual ~EdgeEvaluator() {}
 
   virtual bool evaluate(const Rung_<FloatType>& from,
                         const Rung_<FloatType>& to,
-                        std::vector<typename LadderGraph<FloatType>::EdgeList>& edges) = 0;
+                        std::vector<typename LadderGraph<FloatType>::EdgeList>& edges)
+  {
+    const auto n_start = from.data.size() / dof_;
+    const auto n_end = to.data.size() / dof_;
 
-  typedef typename std::shared_ptr<EdgeEvaluator<FloatType>> Ptr;
+    // Allocate if needed because of Compound Evaluator)
+    if (edges.size() == 0)
+      edges.resize(n_start);
+
+    assert(edges.size() == n_start);
+
+    for (std::size_t i = 0; i < n_start; ++i)
+    {
+      const auto* start_vertex = from.data.data() + dof_ * i;
+      for (std::size_t j = 0; j < n_end; ++j)
+      {
+        const auto* end_vertex = to.data.data() + dof_ * j;
+
+        // Consider the edge:
+        auto results = considerEdge(from, start_vertex, to, end_vertex);
+        if (results.first)
+          edges[i].emplace_back(results.second, j);
+      }
+    }
+
+    for (const auto& rung : edges)
+      if (!rung.empty())
+        return true;
+
+    return false;
+  }
+
+  /**
+   * @brief This evalutes the edge and determines if it should be considered.
+   * @param from The rung associated with the start state
+   * @param start The start state of the edge
+   * @param to The rung associated with the end state
+   * @param end The end state of the edge
+   * @return A pair <True/False, Cost>
```

Comment:
More description -> True if edge is valid, false otherwise. Cost to move from the first vertex to the second